internal/pkg/rpaas/k8s.go

```diff
@@ -755,26 +755,60 @@ func (m *k8sRpaasManager) BindApp(ctx context.Context, instanceName string, args
 		return &ValidationError{Msg: "application host cannot be empty"}
 	}
 
-	if instance.Spec.Host != "" && instance.Spec.Host != args.AppHost {
-		return &ConflictError{Msg: "instance already bound with another application"}
+	if len(instance.Spec.Binds) > 0 {
+		for _, value := range instance.Spec.Binds {
+			if value.Host == args.AppHost {
+				return &ConflictError{Msg: "instance already bound with this application"}
+			}
+		}
+	}
+	if instance.Spec.Binds == nil {
+		instance.Spec.Binds = make([]v1alpha1.Bind, 0)
 	}
 
-	instance.Spec.Host = args.AppHost
+	instance.Spec.Binds = append(instance.Spec.Binds, v1alpha1.Bind{Host: args.AppHost, Name: args.AppName})
 
 	return m.cli.Update(ctx, instance)
 }
 
-func (m *k8sRpaasManager) UnbindApp(ctx context.Context, instanceName string) error {
+func (m *k8sRpaasManager) UnbindApp(ctx context.Context, appName, instanceName string) error {
 	instance, err := m.GetInstance(ctx, instanceName)
 	if err != nil {
 		return err
 	}
 
-	if instance.Spec.Host == "" {
+	if len(instance.Spec.Binds) == 0 {
+		return &ValidationError{Msg: "instance not bound"}
+	} else if instance.Spec.Binds[0].Host == "" {
 		return &ValidationError{Msg: "instance not bound"}
 	}
 
-	instance.Spec.Host = ""
+	if appName == "" {
+		return &ValidationError{Msg: "must specify an app name"}
+	}
+
+	if len(instance.Spec.Binds) == 1 {
+		if appName == instance.Spec.Binds[0].Name {
+			instance.Spec.Binds = []v1alpha1.Bind{}
+			return m.cli.Update(ctx, instance)
+		}
+		return &NotFoundError{Msg: "app not bound to instance"}
+	}
```

Comment:
I don't think you need this check since you do this after the `for` loop on the next lines.